packages/shared/src/hooks/source/useSourceActions.ts

```diff
@@ -0,0 +1,40 @@
+import { Source } from '../../graphql/sources';
+import { useSourceActionsBlock } from './useSourceActionsBlock';
+import { useSourceActionsNotify } from './useSourceActionsNotify';
+import { useSourceActionsFollow } from './useSourceActionsFollow';
+
+interface UseSourceActionsProps {
+  source: Source;
+}
+
+interface UseSourceActionsReturn {
+  isBlocked: boolean;
+  toggleBlock: () => void;
+  isFollowing: boolean;
+  toggleFollow: () => void;
+  haveNotifications: boolean;
```

Comment:
Im not sure of the name is clear enough, it sounds like user has notifications open.
I think you want to say when someone has notify on right?
Not sure whats better but feel this naming might give confusion down the line (cc @nensidosari what do you feel?)

Reply:
Yeah it's a bit confusing, maybe `haveNotificationsOn` is really better.